FactoryWeather/App/VIP Search/SearchInteractor.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  SearchInteractor.swift
+//  FactoryWeather
+//
+//  Created by Matej Korman on 14/01/2019.
+//  Copyright © 2019 Matej Korman. All rights reserved.
+//
+
+import Foundation
+import Promises
+
+protocol SearchBusinessLogic {
+    func getLocations(forText text: String)
+}
+
+class SearchInteractor {
+    var presenter: SearchPresentationLogic?
+    lazy var locationsWorker = LocationsWorker()
+}
+
+// MARK: - Business Logic
+extension SearchInteractor: SearchBusinessLogic {
+    func getLocations(forText text: String) {
+        locationsWorker.getLocations(forText: text)
+            .then { [weak self] locations in self?.presenter?.presentLocations(locations) }
+            .catch { [weak self] error in self?.presenter?.presentError(error) }
```

Comment:
We can shorten this by using anonymous parameters:
`.catch { [weak self] error in self?.presenter?.presentError(error) }`